### ISSUE TITLE
[used-before-assignment] False negative when nested in if statements

### ISSUE THREAD

**Issue by Kakadus** [pylint-dev/pylint#9859]
### Bug description

```python
When both assignment branches and usage is nested inside separate with equivalent checks, a used before assignment is not detected. Pylint should detect these as it does, if there is no guard around assignment and usage.


from enum import Enum


class ABC(Enum):
    A = "A"
    B = "B"
    C = "C"


def true_positive(a: ABC):
    if a == ABC.A:
        c = "user"
    if a == ABC.B:
        c = "contributor"
    print(c)  # true positive


def false_negative(a: ABC, b: bool):
    if b:
        if a == ABC.A:
            c = "user"
        if a == ABC.B:
            c = "contributor"
    # do more stuff
    if b:
        print(c)  # false negative
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:15:10: E0606: Possibly using variable 'c' before assignment (possibly-used-before-assignment)
```

### Expected behavior

l. 26 errors as well, if `a == ABC.C`, so the output would be smth like
```
************* Module test
test.py:15:10: E0606: Possibly using variable 'c' before assignment (possibly-used-before-assignment)
test.py:26:14: E0606: Possibly using variable 'c' before assignment (possibly-used-before-assignment)


### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522]
```

### OS / Environment

Arch Linux

**Comment by jacobtylerwalls** [comment]
Hi @kakadus. This little false negative is currently intentional and [described here](https://pylint.readthedocs.io/en/latest/user_guide/messages/error/possibly-used-before-assignment.html) (see the discussion under `if guarded()`.

The question is, should we remove it? We didn't want to spam projects with unwelcome messages when `# do more stuff` has no impact on `b`, but you present here a case where we should emit.

**Comment by Kakadus** [comment]
I admit I did not read that bit in the docs. I just noticed that my PyCharm IDE highlighted this, but pylint were silent. I'd preferred if pylint warns about this case. However, I cannot assess the spamming part here...
